src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs

```diff
@@ -851,6 +887,65 @@ public void Type_Attribute_Delete_NotSupportedByRuntime2()
                 Diagnostic(RudeEditKind.ChangingAttributesNotSupportedByRuntime, "class C", FeaturesResources.class_));
         }
 
+        [Fact]
+        public void Type_Attribute_Change_Reloadable()
+        {
+            var attributeSrc = @"
+public class A1 : System.Attribute { }
+public class A2 : System.Attribute { }
+public class A3 : System.Attribute { }
+";
+
+            var src1 = ReloadableAttributeSrc + attributeSrc + "[CreateNewOnMetadataUpdate, A1, A2]class C { }";
+            var src2 = ReloadableAttributeSrc + attributeSrc + "[CreateNewOnMetadataUpdate, A2, A3]class C { }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [[CreateNewOnMetadataUpdate, A1, A2]class C { }]@267 -> [[CreateNewOnMetadataUpdate, A2, A3]class C { }]@267");
+
+            edits.VerifySemantics(
+                SemanticEdit(SemanticEditKind.Replace, c => c.GetMember("C")));
+        }
+
+        [Fact]
+        public void Type_Attribute_ReloadableRemove()
+        {
+            var src1 = ReloadableAttributeSrc + "[CreateNewOnMetadataUpdate]class C { }";
+            var src2 = ReloadableAttributeSrc + "class C { }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemantics(
+                SemanticEdit(SemanticEditKind.Replace, c => c.GetMember("C")));
+        }
+
+        [Fact]
+        public void Type_Attribute_ReloadableAdd()
+        {
+            var src1 = ReloadableAttributeSrc + "class C { }";
+            var src2 = ReloadableAttributeSrc + "[CreateNewOnMetadataUpdate]class C { }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemantics(
+                ActiveStatementsDescription.Empty,
+                new[] { SemanticEdit(SemanticEditKind.Update, c => c.GetMember("C")) },
+                capabilities: EditAndContinueTestHelpers.Net6RuntimeCapabilities);
```

Comment:
It's surprising to me that this wouldn't be a Reload, if the test above is. Not a big deal though I guess.

Reply:
It's not since the original type is not reloadable. The next edit would be reload.

Reply:
In the above test the original type is reloadable hence the edit is reload. The next edit in that case would be in-place update.